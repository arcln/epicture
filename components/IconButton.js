import React from 'react';
import {TouchableOpacity} from 'react-native';
import {Icon} from 'expo';
import Colors from '../constants/Colors';

export default class IconButton extends React.Component {
  render() {
    return (
      <TouchableOpacity onPress={this.props.onPress} style={{backgroundColor: 'transparent'}}>
        <Icon.Ionicons
          name={this.props.name}
          style={{color: this.props.color || Colors.tintColor}}
          size={this.props.size || 24}
        />
      </TouchableOpacity>
    );
  }
}
