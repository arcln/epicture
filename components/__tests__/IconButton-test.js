import {View, Platform} from 'react-native';
import React from 'react';
import IconButton from '../IconButton';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(
    <View>
      <IconButton
        size={100}
        color='#123456'
        name={Platform.OS === 'ios' ? 'ios-eye' : 'md-eye'}
      />
      <IconButton
        size={40}
        color='#123456'
        name={Platform.OS === 'ios' ? 'ios-eye' : 'md-eye'}
      />
      <IconButton
        size={20}
        color='#123456'
        name={Platform.OS === 'ios' ? 'ios-eye' : 'md-eye'}
      />
      <IconButton
        size={10}
        color='#123456'
        name={Platform.OS === 'ios' ? 'ios-eye' : 'md-eye'}
      />
      <IconButton
        size={8}
        color='#123456'
        name={Platform.OS === 'ios' ? 'ios-eye' : 'md-eye'}
      />
    </View>
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
