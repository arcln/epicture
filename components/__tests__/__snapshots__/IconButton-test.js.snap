// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "backgroundColor": "transparent",
        "opacity": 1,
      }
    }
  >
    <Text />
  </View>
</View>
`;
